rust/src/builders/witness_builder.rs

```diff
@@ -338,15 +338,41 @@ impl TransactionWitnessSetBuilder {
             InputAggregateWitnessData::PlutusScriptNoDatum(witness, info) => {
                 self.add_plutus_script(&witness.script());
                 self.add_plutus_datum(&witness.untagged_redeemer().datum());
-                self.add_fake_vkey_witnesses_by_num(info.missing_signers.len());
-                self.add_fake_vkey_witnesses(&info.known_signers.0);
+
+                let mut required_wits = RequiredWitnessSet::new();
```

Comment:
Missing signers are deduplicated after the key hashes are given to `required_wits`. This [code](https://github.com/dcSpark/cardano-multiplatform-lib/pull/57/files#diff-0318c1c246c0c5d8f74f1b087e11b7e27f27f353c2443ecc886914dca673db44R354) should be able to resolve multiple calls of `add_input_aggregate_witness_data`.

Reply:
The `required_wits` is updated [here](https://github.com/dcSpark/cardano-multiplatform-lib/pull/57/files#diff-0318c1c246c0c5d8f74f1b087e11b7e27f27f353c2443ecc886914dca673db44R351).

Reply:
A related test is added [here](https://github.com/dcSpark/cardano-multiplatform-lib/pull/57/files#diff-0318c1c246c0c5d8f74f1b087e11b7e27f27f353c2443ecc886914dca673db44R508).

Reply:
> Additionally, we don't need fake witnesses for known_signers because we know what the real key is.

Still need to calculate the size for fee, right? Otherwise where is supposed to fetch the real witness?